### PR TITLE
[R-package] Updated lgb.train.R with keyword arguments

### PR DESCRIPTION
**Pull request by iadi7ya**
Fixes lgb.train.R from a list of functions listed on #3390

### CODE REVIEW

**Comment by iadi7ya** [comment]
Hi @jameslamb. Can you please regenerate the docs and guide me how can I run the checks again. Thanks:)

File: R-package/R/lgb.train.R [hunk 1]
```diff
@@ -47,7 +47,7 @@
 #' }
 #' @export
 lgb.train <- function(params = list(),
-                      data,
+                      data = data,
```
Comment (jameslamb) [inline R-package/R/lgb.train.R:50]:
```suggestion
                      data,
```

Can you please revert this? This one is not appropriate, since it would be getting an object called `data` from the calling environment. No object like that is guaranteed to exist.

File: R-package/R/lgb.train.R [hunk 2]
```diff
@@ -238,7 +238,7 @@ lgb.train <- function(params = list(),
   # If user supplied early_stopping_rounds, add the early stopping callback
   if (using_early_stopping) {
     callbacks <- add.cb(
-      callbacks
+      callbacks = callbacks
```
Comment (jameslamb) [inline R-package/R/lgb.train.R:241]:
```suggestion
      cb_list = callbacks
```

This is not the correct argument. Please see the definition in https://github.com/microsoft/LightGBM/blob/785eb60c7864f195230d70bc817f53feb6d46a81/R-package/R/callback.R#L409. Can you please change this to `cb_list`, and add the keyword argument `cb` on the line right below it?

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thank you for the contribution! I've requested a few changes to this PR. In addition, can you please also add the appropriate keyword arguments for the following calls?

* (line 250) `categorize.callbacks()`
* (line 255) `booster$set_train_data_name()`
* (line 259) `booster$add_valid()`

File: R-package/R/lgb.train.R [hunk 3]
```diff
@@ -82,10 +82,10 @@ lgb.train <- function(params = list(),
 
   # Setup temporary variables
   additional_params <- list(...)
-  params <- append(params, additional_params)
+  params <- append(params = params, additional_params = additional_params)
```
Comment (jameslamb) [inline R-package/R/lgb.train.R:85]: these are not the keyword arguments to `append()`. You can find the arguments by running `?append` in an R session.

Please revert this change. You can leave `append()` calls unchanged, since they are R built-ins.

```suggestion
  params <- append(params, additional_params)
```

**Comment by jameslamb** [comment]
> Hi @jameslamb. Can you please regenerate the docs and guide me how can I run the checks again. Thanks:)

thanks so much for the pull request! I've left some requested changes. Once those have been made, I can come update the documentation.

**Comment by iadi7ya** [comment]
Hi @jameslamb I have fixed the changes which you have suggested. Most of the checks have passed but in 3 of the checks I am getting following error:
```
The downloaded binary packages are in C:\Users\runneradmin\AppData\Local\Temp\RtmpuoWnIc\downloaded_packages
Downloading https://github.com/microsoft/LightGBM/releases/download/v2.0.12/miktexsetup-4.0-x64.zip
Setting up MiKTeX
miktexsetup.exe: The remote package repository is outdated. You have to choose another repository.
miktexsetup.exe: Data: url="https://ctan.math.illinois.edu/systems/win32/miktex/tm/packages/"
```


Is that a problem or is it ready for merge?

**Comment by jameslamb** [comment]
this happens sometimes and is definitely unrelated to your changes. I just restarted the builds, hopefully they'll pass.

**Comment by jameslamb** [comment]
It looks like we have an ongoing issue with the MikTeX ecosystem (#3454 ). I'll merge this PR once I'm able to fix that, apologies.

**Comment by iadi7ya** [comment]
Thanks @jameslamb :)

**Review by jameslamb — APPROVED** [review]
Ok now the tests are passing!

Thanks very much for contributing, @iadi7ya . I hope you'll consider contributing more in the future :grinning:

**Comment by iadi7ya** [comment]
That's awesome @jameslamb ! I'd love to keep contributing to this wonderful project :)
